sw/device/lib/dif/meson.build

```diff
@@ -725,8 +725,29 @@ sw_lib_dif_sram_ctrl = declare_dependency(
   link_with: static_library(
     'sw_lib_dif_sram_ctrl',
     sources: [
+      hw_ip_sram_ctrl_reg_h,
       'dif_sram_ctrl.c',
     ],
     dependencies: [sw_lib_mmio],
```

Comment:
need to add the autogen'd target, which you should also define in `sw/device/lib/dif/autogen/meson.build` (the meson.build in `sw/device/lib/dif/autogen/meson.build` is not autogen'd yet, though it could, and probably will, be in the future)